sdcm/cluster_cloud.py

```diff
@@ -0,0 +1,483 @@
+
+        self._public_ip = cloud_instance_data.get('publicIp')
+        self._private_ip = cloud_instance_data.get('privateIp')
+
+        name = f"{node_prefix}-{dc_idx}-{node_index}".lower()
+        super().__init__(
+            name=name,
+            parent_cluster=parent_cluster,
+            base_logdir=base_logdir,
+            node_prefix=node_prefix,
+            dc_idx=dc_idx,
+            rack=rack
+        )
+
+    def __str__(self):
+        return f'CloudNode {self.name} [{self.public_ip_address} | {self.private_ip_address}]{self._dc_info_str()}'
+
+    @property
+    def network_interfaces(self):
+        return [{
+            'public_ip': self._public_ip,
+            'private_ip': self._private_ip,
+            'interface_name': 'eth0'
+        }]
+
+    def _refresh_instance_state(self):
+        try:
+            node_details = self._api_client.get_cluster_nodes(
+                account_id=self._account_id,
+                cluster_id=self._cluster_id)
+            for node_data in node_details:
+                if node_data.get('id') == self._node_id:
+                    self._cloud_instance_data = node_data
+                    self._public_ip = node_data.get('publicIp', self._public_ip)
+                    self._private_ip = node_data.get('privateIp', self._private_ip)
+                    break
+
+            return [self._public_ip], [self._private_ip]
+        except Exception as e:  # noqa: BLE001
+            self.log.warning("Failed to refresh instance state: %s", e)
+            return [self._public_ip], [self._private_ip]
+
+    @property
+    def public_ip_address(self):
+        return self._public_ip
+
+    @property
+    def private_ip_address(self):
+        return self._private_ip
+
+    @property
+    def ipv6_ip_address(self):
+        return None
+
+    @property
+    def vm_region(self):
+        return self._cloud_instance_data.get('region', 'unknown')
+
+    @property
+    def region(self):
+        return self.vm_region
+
+    @property
+    def datacenter(self):
+        return f"datacenter{self.dc_idx + 1}"
+
+    def _get_ipv6_ip_address(self):
+        return None
+
+    def wait_for_cloud_init(self):
+        self.log.debug("Skip waiting for cloud-init on scylla-cloud, no approach to SSHing nodes for now")
+
+    def _init_port_mapping(self):
+        pass
+
+    def set_keep_alive(self):
+        # only log for now for PoC
+        self.log.info("Setting keep alive for node %s", self.name)
+        return True
+
+    def restart(self):
+        # there is no dedicated Scylla Cloud API for node restart;
+        # this should be implemented after approach to accessing cloud nodes is defined
+        pass
+
+    def terminate(self):
+        # there is no dedicated Scylla Cloud API for node termination;
+        # it should be handled at the cluster level through resizing
+        self.log.warning("Individual node termination is not supported in Scylla Cloud. "
```

Comment:
I would recommend raising exception here

we should know it's not working when we get to it, warning in the log would be enough.